app/src/main/java/com/android/solvit/provider/ui/request/ListRequestsFeed.kt

```diff
@@ -0,0 +1,434 @@
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.items
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.foundation.shape.RoundedCornerShape
+import androidx.compose.material.icons.Icons
+import androidx.compose.material.icons.filled.Menu
+import androidx.compose.material.icons.filled.MoreVert
+import androidx.compose.material.icons.filled.Notifications
+import androidx.compose.material3.DropdownMenu
+import androidx.compose.material3.DropdownMenuItem
+import androidx.compose.material3.Icon
+import androidx.compose.material3.IconButton
+import androidx.compose.material3.OutlinedTextField
+import androidx.compose.material3.Scaffold
+import androidx.compose.material3.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.MutableState
+import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.graphics.ColorFilter
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.platform.testTag
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.TextStyle
+import androidx.compose.ui.text.font.FontWeight
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.compose.ui.unit.sp
+import androidx.lifecycle.viewmodel.compose.viewModel
+import coil.compose.AsyncImage
+import com.android.solvit.R
+import com.android.solvit.seeker.ui.navigation.SeekerBottomNavigationMenu
+import com.android.solvit.shared.model.map.Location
+import com.android.solvit.shared.model.request.ServiceRequest
+import com.android.solvit.shared.model.request.ServiceRequestViewModel
+import com.android.solvit.shared.model.service.Services
+import com.android.solvit.shared.ui.map.GetDirectionsBubble
+import com.android.solvit.shared.ui.navigation.LIST_TOP_LEVEL_DESTINATION_PROVIDER
+import com.android.solvit.shared.ui.navigation.NavigationActions
+import java.text.SimpleDateFormat
+import java.util.Locale
+
+@Preview
+@Composable
+fun RequestsTopBar() {
+  val context = LocalContext.current
+  Row(
+      modifier = Modifier.fillMaxWidth().background(Color.White).testTag("RequestsTopBar"),
+      horizontalArrangement = Arrangement.SpaceBetween,
+      verticalAlignment = Alignment.CenterVertically) {
+        IconButton(
+            modifier = Modifier.testTag("MenuOption"),
+            onClick = {
+              Toast.makeText(context, "Not Yet Implemented", Toast.LENGTH_LONG).show()
```

Comment:
Nit: This is more of a global comment and should be a new task for this sprint on the entire project. The Toast could be placed in a file of reusables in a function that could have a few implementations pre-defined: "Not Yet Implemented", "Login Successful", "Login failed", etc as 
```
fun showToast(context: Context, string: String) {
    Toast.makeText(context, string, Toast.LENGTH_LONG).show()
}

fun showNotImplementedToast(context: Context) {
    showToast(context, "Not Yet Implemented") // And even better would be R.string.not_yet_implemented 
}
```

Reply:
That's a good remark! As I said before, we could create a more general task to regroup every component which is used several times throughout the project in a single file.